### PR TITLE
VIM-5585: Update VIMLive and VIMVideo

### PR DESCRIPTION
**Pull request by jasonhawkins**
#### Ticket

[VIM-5585](https://vimean.atlassian.net/browse/VIM-5585)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Implementation Summary
- Added constants to VIMLive to mirror the status enum, because the enum is not available via Objective-C.
- Added convenience methods to return whether or not a live video is pre, mid, or post broadcast.

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/b63f4e388937bb3efe9fab886dcc4aba40603c5e/VimeoNetworking/Sources/VimeoClient.swift#L331">VimeoNetworking/Sources/VimeoClient.swift#L331</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 62 tests, with 0 failures (0 unexpected) in 0.715 (0.748) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/488)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoNetworking/Sources/Models/VIMVideo.m [hunk 1]
```diff
@@ -506,4 +506,19 @@ - (BOOL)isLive
...
+}
+
+- (BOOL)isMidBroadcast
+{
+    return self.isLive && [self.live.status isEqual: VIMLive.LiveStreamStatusStreaming];
+}
+
+- (BOOL)isPostBroadcast
+{
+    return self.isLive && ([self.live.status isEqual: VIMLive.LiveStreamStatusDone] || [self.live.status isEqual: VIMLive.LiveStreamStatusStreamingError]);
```
Comment (supperspidey) [inline VimeoNetworking/Sources/Models/VIMVideo.m]: Per our conversation, let's make sure that checking for `streaming_error` is relevant for post-broadcasting.

File: VimeoNetworking/Sources/Models/VIMLive.swift [hunk 2]
```diff
@@ -50,6 +50,14 @@ public enum LiveStreamingStatus: String
 /// a `clip` response.
 public class VIMLive: VIMModelObject
 {
+    public static let LiveStreamStatusUnavailable = "unavailable"
```
Comment (supperspidey) [inline VimeoNetworking/Sources/Models/VIMLive.swift]: Let's have documentation for those constants too.

File: VimeoNetworking/Sources/Models/VIMVideo.m (hunk 1 above)
Comment (jasonhawkins) [reply VimeoNetworking/Sources/Models/VIMVideo.m]: Ah, I see what you're saying now. That sounds good.

**Comment by jasonhawkins** [comment]
Thanks @supperspidey, ready for another look.

**Comment by supperspidey** [comment]
Good to go @jasonhawkins

**Pull request merged**